src/Node/FS/Async.purs

```diff
@@ -333,3 +349,91 @@ exists :: forall eff. FilePath
                    -> Eff (fs :: FS | eff) Unit
 exists file cb = mkEff $ \_ -> runFn2
   fs.exists file $ \b -> runPure (unsafeInterleaveEff (cb b))
+
+
+{- Asynchronous File Descriptor Functions -}
+
+--|
+-- Open a file asynchronously.  See <a
+-- href="https://nodejs.org/api/fs.html#fs_fs_open_path_flags_mode_callback">Node
+-- Documentation</a> for details.
+--
+fdOpen :: forall eff.
+          FilePath
+       -> FileFlags
+       -> Maybe FileMode
+       -> Callback eff FileDescriptor
+       -> Eff (fs :: FS | eff) Unit
+fdOpen file flags mode cb =
+  case mode of
+    Nothing  -> mkEff $ \_ -> runFn3 fs.open file (show flags) (handleCallback cb)
+    (Just m) -> mkEff $ \_ -> runFn4 create file (show flags) m (handleCallback cb)
+
+--|
+-- Read from a file asynchronously.  See <a
+-- href="https://nodejs.org/api/fs.html#fs_fs_read_fd_buffer_offset_length_position_callback">Node
+-- Documentation</a> for details.
+--
+fdRead :: forall eff.
+          FileDescriptor
+       -> Buffer
+       -> BufferOffset
+       -> BufferLength
+       -> Maybe FilePosition
+       -> Callback eff ByteCount
+       -> Eff (fs :: FS | eff) Unit
+fdRead fd buff off len Nothing cb =
```

Comment:
This looks like it would be better handled by `purescript-nullable`. Then we wouldn't need to import two different versions using the FFI. Same for `writeSeq`. @garyb does that sound ok?